src/resources/nfts/index.ts

```diff
@@ -99,3 +115,24 @@ export function useLegacyNFTs({ address }: { address: string }) {
     isInitialLoading: !nfts.length && isFetching,
   };
 }
+
+export function useNFTListing({
+  contractAddress,
+  tokenId,
+  network,
+}: {
+  contractAddress: string;
+  tokenId: string;
+  network: Omit<Network, Network.goerli>;
+}) {
+  return useQuery(
+    nftListingQueryKey({ contractAddress, tokenId, network }),
+    async () =>
+      (await fetchSimpleHashNFTListing(contractAddress, tokenId, network)) ??
+      null,
+    {
+      enabled: !!network && !!contractAddress && !!tokenId,
```

Comment:
Just a general note for @rainbow-me/app-team 

This `enabled` prop is totally necessary rn and we need to use it here. But keep in mind _why_ we need this type of thing in the codebase rn 🤔 it's because we _don't know when/if we have data_.

What we want are stronger guarantees around our data. In a future-utopian world, we'd fetch data at a higher level, wait for it to load, and then consume safely within components further down the component tree. In that world, we'd know that we already have these values ready for us, and wouldn't need `enabled`.

There are exceptions to everything — `enabled` will probably be useful in other ways — but when thinking about data fetching moving forward, this is a good model to shoot for.